### ISSUE TITLE
inconsistent TRUNCATE behaviour after changing citus. replication_model

### ISSUE THREAD

**Issue by onurctirtir** [citusdata/citus#3516]
Assuming the mx case and the following tables involved in a foreign key relationship,

```sql
-- start metadata sync to workers
SELECT start_metadata_sync_to_node(<worker1Host>, <worker1Port>); 
SELECT start_metadata_sync_to_node(<worker2Host>, <worker2Port>); 
-- however, do not set replication model to streaming here to
-- reproduce the buggy case

-- we could also use a regular dist. table instead of a reference table as 
-- referenced table, same inconsistency applies to that case too
create table ref_table(a int primary key);
select create_reference_table('ref_table');

create table dist_table(a int);
select create_distributed_table('dist_table', 'a');

alter table dist_table add constraint fkey foreign key (a) references ref_table(a);

-- now set the replication model
ALTER SYSTEM SET citus.replication_model TO 'streaming';
SELECT pg_reload_conf();
```
, when we execute below commands from the coordinator and from an mx worker, we get following inconsistent error messages related with `CASCADE` behavior. 

```sql
-- on coordinator: -----------------------
-- coordinator behavior is actually the expected one from PorsgreSQL perspective
postgres=# truncate ref_table ;
ERROR:  cannot truncate a table referenced in a foreign key constraint
DETAIL:  Table "dist_table" references "ref_table".
HINT:  Truncate table "dist_table" at the same time, or use TRUNCATE ... CASCADE.

-- on an mx worker: -----------------------
-- we get no errors here. not the expected behavior
-- as worker is not even aware of the fkey constraint defined on `ref_table`
postgres=# truncate ref_table ;
TRUNCATE TABLE
```

**Comment by onderkalaci** [comment]
I cannot reproduce this on master, I get the same error on the workers. Which branch/release can you reproduce this?

**Comment by onurctirtir** [comment]
> I cannot reproduce this on master, I get the same error on the workers. Which branch/release can you reproduce this?

on master

**Comment by onurctirtir** [comment]
The exact case is setting up the `replication_model` to `streaming`  after creatating / distributing tables. , which is expected to be done at the very early times when we create an mx cluster normally. 

If we set the replication model to streaming after creating / distributing the tables, some inconsistencies happen. Workers in this case do not even know the reference table is referenced via a distributed table (without knowing the exact reason). 

I will investigate the issue and open a pr for it after developing a solution

**Comment by onurctirtir** [comment]
Keeping the issue open for further improvements on "setting replication model". No solutions needed to be develeoped in this case as we rarely do such operations (at the very early times setting up the cluster)

**Comment by hanefi** [comment]
Closed by #4891 that deprecated the `replication_model` GUC.
